packages/browser-wallet/src/popup/popupX/shared/IdCard/IdCard.tsx

```diff
@@ -2,35 +2,64 @@ import React, { ReactNode } from 'react';
 import Card from '@popup/popupX/shared/Card';
 import Text from '@popup/popupX/shared/Text';
 import Button from '@popup/popupX/shared/Button';
+import useEditableValue from '@popup/popupX/shared/EditableValue';
 
-interface IdCardProps {
-    rowsIdInfo?: [string, string | ReactNode][];
-    rowsConnectedAccounts?: [string, string][];
-    onEditName?: () => void;
-}
+export type IdCardAttributeInfo = {
+    key: string;
+    value: string | ReactNode;
+};
+
+export type IdCardAccountInfo = {
+    address: string;
+    amount: ReactNode;
+};
+
+export type IdCardProps = {
+    idProviderName: string;
+    identityName: string;
+    rowsIdInfo?: IdCardAttributeInfo[];
+    rowsConnectedAccounts?: IdCardAccountInfo[];
+    onNewName?: (newName: string) => void;
+    identityNameFallback?: string;
+};
 
-export default function IdCard({ rowsIdInfo = [], rowsConnectedAccounts, onEditName }: IdCardProps) {
+export default function IdCard({
+    idProviderName,
+    identityName,
+    rowsIdInfo = [],
+    rowsConnectedAccounts,
+    onNewName,
+    identityNameFallback,
+}: IdCardProps) {
+    const editable = useEditableValue(identityName, identityNameFallback ?? '', onNewName ?? (() => {}));
     return (
         <Card type="gradient" className="id-card-x">
             <span className="title-row">
-                <Text.Main>Identity 1</Text.Main>
-                {onEditName && <Button.Secondary label="Edit Name" />}
+                <Text.Main>{editable.value}</Text.Main>
+                {editable.isEditing ? (
+                    <>
+                        <Button.Secondary label="Save" onClick={editable.onComplete} />
+                        <Button.Secondary label="Abort" onClick={editable.onAbort} />
+                    </>
+                ) : (
+                    onNewName && <Button.Secondary label="Edit Name" onClick={editable.onEdit} />
+                )}
```

Comment:
It itches in an annoying place in my brain when the buttons to edit and save/abort are not the same as for the account.. But I guess this is what the design specifies 🤷

Reply:
The design only contains the `Edit name` button, the `Save` and `Abort` are me improvising.
But I agree, it is not consistent.

Reply:
Yeah.. let's just keep this then.. It's a minor thing anyway